vote-sc/src/views.rs

```diff
@@ -0,0 +1,76 @@
+multiversx_sc::imports!();
+
+use crate::{
+    constants::{Hash, ProposalId, HASH_LENGTH, PROOF_LENGTH},
+    errors::NO_PROPOSAL,
+};
+
+#[multiversx_sc::module]
+pub trait ViewsModule {
+    #[view(getProposalRootHash)]
+    fn get_root_hash(
+        &self,
+        proposal_id: ProposalId,
+    ) -> OptionalValue<ManagedByteArray<HASH_LENGTH>> {
+        if self.root_hash_proposal_nonce(proposal_id).is_empty() {
+            return OptionalValue::None;
+        }
+
+        OptionalValue::Some(self.root_hash_proposal_nonce(proposal_id).get())
+    }
+
+    #[view(confirmVotingPower)]
+    fn confirm_voting_power(
+        &self,
+        proposal_id: ProposalId,
+        voting_power: BigUint<Self::Api>,
+        proof: ArrayVec<ManagedByteArray<HASH_LENGTH>, PROOF_LENGTH>,
+    ) -> bool {
+        match self.get_root_hash(proposal_id) {
+            OptionalValue::None => {
+                sc_panic!(NO_PROPOSAL);
+            }
+            OptionalValue::Some(root_hash) => {
+                self.verify_merkle_proof(voting_power, proof, root_hash)
+            }
+        }
+    }
+
+    fn verify_merkle_proof(
```

Comment:
This was the required implementation? Can't you use signatures that you can check? Like it is used in the Metabonding SC?

Reply:
this is just copied from `governance_v2`. I had nothing to do with how this thing works